child_compassion/model/child_pictures.py

```diff
@@ -109,8 +109,18 @@ def create(self, cr, uid, vals, context=None):
             # Attach the picture to the last Case Study
             case_study = child.case_study_ids and child.case_study_ids[0]
             if case_study and not case_study.pictures_id:
-                case_study.attach_pictures(res_id)
-                child_picture.write({'case_study_id': case_study.id})
+                six_months = 180
+                case_study_date = case_study.info_date
+                case_study_date = datetime.strptime(case_study_date, DF)
+
+                picture_date = child_picture.date
+                picture_date = datetime.strptime(picture_date, DF)
+
+                date_diff = abs((case_study_date - picture_date).days)
+
+                if (date_diff <= six_months):

```

Comment:
You can add condition or child.type == 'LDP' so that we link case study of LDP child (the date is not important for LDP since they don't have a real Case Study).